core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -339,4 +349,9 @@ public boolean matchesIndices(String... indices) {
         }
         return false;
     }
+
+    public List<Object> handleTermsLookup(TermsLookup termsLookup) {
+        termsLookup.setIndex((termsLookup.getIndex() == null) ? this.index().name() : termsLookup.getIndex());
```

Comment:
if the index is null, then get it from the context. Sure we do mutate the object, should we instead operate on a cloned object? I would much prefer keeping this logic where the action happens.